app/javascript/packs/webauthn-setup.js

```diff
@@ -0,0 +1,57 @@
+function webauthn() {
+  const arrayBufferToBase64 = function(buffer) {
+    let binary = '';
+    const bytes = new Uint8Array(buffer);
+    const len = bytes.byteLength;
+    for (let i = 0; i < len; i += 1) {
+      binary += String.fromCharCode(bytes[i]);
+    }
+    return window.btoa(binary);
+  };
+  const longToByteArray = function(long) {
+    const byteArray = new Uint8Array(8);
+    for (let index = 0; index < byteArray.length; index += 1) {
+      const byte = long & 0xff;  // eslint-disable-line no-bitwise
+      byteArray[index] = byte;
+      long = (long - byte) / 256;
+    }
+    return byteArray;
+  };
+  const userId = document.getElementById('user_id').value;
+  const userEmail = document.getElementById('user_email').value;
+  const challengeBytes = new Uint8Array(JSON.parse(document.getElementById('user_challenge').value));
+  const createOptions = {
+    publicKey: {
+      challenge: challengeBytes,
+      rp: { name: window.location.hostname },
+      user: {
+        id: longToByteArray(userId),
+        name: userEmail,
+        displayName: userEmail,
+      },
+      pubKeyCredParams: [
```

Comment:
We should probably work with our security folk to come up with a final list of algorithms. These are good enough to get us up and running during development.

Reply:
@commit-dkp Any thoughts here? Do you want to help us work out the algorithms to use for this?

Reply:
Just noting for the record, `alg -7` aka `ES256` aka `ECDSA w/ SHA-256` in https://www.iana.org/assignments/cose/cose.xhtml is a fine algorithm and the one that we can expect every FIDO certified device to support.

WebAuthN aka `FIDO2` encompasses many more algorithms to choose from, but the only one I'd recommend adding at this point is `alg -8` aka `EdDSA`.  However, as `EdDSA` is not yet embraced by NIST, the next best algorithm to prefer over `alg -7` would be `alg -35` aka `ES384` aka `ECDSA w/SHA-384`.

This way, if we ever have reason to remove support for either one, we will still have had the other in use for some time and can continue to depend on it while we decide on what to replace the removed algorithm with.

Reply:
Great advice, thank you. @stevegsa @jgsmith-usds Can we include this in any security review of our WebAuthN implementation, and discuss implementing it?

Reply:
I'll add it to our security review/doc and make sure we get it in the next release cycle.

Reply:
@jgsmith-usds actually already did this this morning! https://cm-jira.usa.gov/browse/LG-696 Sorry for the unnecessary ping.